### PR TITLE
Bug 1138372 - [contacts] Contacts app needs to register for text/directo...

### PR DESCRIPTION
**Pull request by adrm**
...ry MIME type for its open activity. (vcard attachments received from outlook/iphone 5 clients may not be opened by mail app while you can open them from Downloads)

### CODE REVIEW

**Comment by try-server-hook** [comment]
[ADLR-es Adrián de la Rosa (ADLR-es)](https://github.com/ADLR-es) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=50cd9787dcf1)

File: apps/communications/contacts/js/activities.js [hunk 1]
```diff
@@ -174,7 +186,16 @@ var ActivityHandler = {
       return;
     }
 
-    if (this.activityDataType.indexOf('text/vcard') !== -1) {
+    // Was this a request for vCard export as a blob?  Check all supported MIME
+    // types.
+    var isVcardDataType = VCARD_MIME_TYPES.some((mime) => {
+      return this.activityDataType.indexOf(mime) != -1;
```
Comment (jmcanterafonseca) [inline apps/communications/contacts/js/activities.js]: nit: !==

**Comment by jmcanterafonseca** [comment]
we would need a test as well

**Comment by try-server-hook** [comment]
[ADLR-es Adrián de la Rosa (ADLR-es)](https://github.com/ADLR-es) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=894817f74d00)

File: apps/communications/contacts/test/unit/contacts_activities_test.js [hunk 2]
```diff
@@ -121,6 +120,33 @@ suite('Test Activities', function() {
...
 
+    test('Open vCard with deprecated datatypes', function() {
+      var activity1 = {
+        source: {
+          name: 'open',
+          data: {
+            type: ['text/directory']
+          }
+        }
+      };
```
Comment (jmcanterafonseca) [inline apps/communications/contacts/test/unit/contacts_activities_test.js]: separate the two legacy MIME types in two separate tests

File: apps/communications/contacts/test/unit/contacts_activities_test.js [hunk 3]
```diff
@@ -121,6 +120,33 @@ suite('Test Activities', function() {
       assert.equal(ActivityHandler._currentActivity, activity);
     });
 
+    test('Open vCard with deprecated datatypes', function() {
```
Comment (jmcanterafonseca) [inline apps/communications/contacts/test/unit/contacts_activities_test.js]: change the title to the referred MIME type tested

**Comment by try-server-hook** [comment]
[ADLR-es Adrián de la Rosa (ADLR-es)](https://github.com/ADLR-es) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2d4c7acfca12)

**Pull request merged**